src/appleseed/renderer/kernel/lighting/lighttree.cpp

```diff
@@ -281,10 +297,17 @@ float LightTree::compute_node_probability(
     // Calculate probability of a single node based on its distance
     // to the surface point being illuminated.
     // For leaf nodes use the actual position of the light source, instead
-    // of center of the bbox. It is more correct for shaped lights.
-    const Vector3d position = node.is_leaf()
-        ? m_light_sources[m_items[node.get_item_index()].m_light_source_index]->get_position()
-        : bbox.center();
+    // of center of the bbox. It is more precise for shaped lights.
+    Vector3d position;
+    if (node.is_leaf() && m_items[node.get_item_index()].m_light_type == EmittingTriangleType)
+    {
+        const size_t light_index = m_items[node.get_item_index()].m_light_index;
+        const EmittingTriangle triangle = m_emitting_triangles[light_index];
+        position = (triangle.m_v0 + triangle.m_v1 + triangle.m_v2) / 3;
```

Comment:
Multiply by `(1.0 / 3.0)` (with the parentheses) instead of dividing by 3. It's not an optimization the compiler is allowed to do.